h2o-persist-s3/src/main/java/water/persist/PersistS3.java

```diff
@@ -35,13 +39,41 @@
 
   private static final String KEY_PREFIX = "s3://";
   private static final int KEY_PREFIX_LEN = KEY_PREFIX.length();
+  private static final int LOCK_TIMEOUT_SEC = 60;
 
-  private static final Object _lock = new Object();
+  private static final Lock _lock = new ReentrantLock();
+  
   private static volatile AmazonS3 _s3;
 
+  public static void changeClientCredentials(final String accesKeyId, final String accessSecretKey) {
```

Comment:
hmm, can we just throw out an existing instance of PersistS3 and start with a new one? keeping this consistent seems error-prone

just thinking out loud, don't take it too seriously

Reply:
We definitely can not, that is why there is some degree of locking introduced. The timeout is there for a simple reason: The call must fail if the new client is not created withing a reasonable period of time, so Flow or other clients don't get stuck on a single call.